amd/src/acceptmodal.js

```diff
@@ -0,0 +1,222 @@
+
+
+/**
+ * Add a create new group modal to the page.
+ *
+ * @module     tool_policy/acceptmodal
+ * @class      AcceptOnBehalf
+ * @package    tool_policy
+ * @copyright  2018 Marina Glancy
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+define(['jquery', 'core/str', 'core/modal_factory', 'core/modal_events', 'core/fragment', 'core/ajax', 'core/yui'],
+    function($, Str, ModalFactory, ModalEvents, Fragment, Ajax, Y) {
+
+        /**
+         * Constructor
+         *
+         * @param {int} contextid
+         *
+         * Each call to init gets it's own instance of this class.
+         */
+        var AcceptOnBehalf = function(contextid) {
+            this.contextid = contextid;
+
+            var stringKeys = [
+                {
+                    key: 'consentdetails',
+                    component: 'tool_policy'
+                },
+                {
+                    key: 'iagreetothepolicy',
+                    component: 'tool_policy'
+                }
+            ];
+            this.strings = Str.get_strings(stringKeys);
+
+            this.init();
+        };
+
+        /**
+         * @var {Modal} modal
+         * @private
+         */
+        AcceptOnBehalf.prototype.modal = null;
+
+        /**
+         * @var {int} contextid
+         * @private
+         */
+        AcceptOnBehalf.prototype.contextid = -1;
+
+        /**
+         * @var {Promise}
+         * @private
+         */
+        AcceptOnBehalf.prototype.strings = 0;
```

Comment:
It feels weird to init a `Promise` variable with zero value. Is that intentional?

Reply:
I copied most of the code from dataprivacy plugin, there are some things I didn't evaluate much